packages/core/src/indexing/database-index-api.ts

```diff
@@ -81,6 +88,25 @@ export abstract class DatabaseIndexApi<DateType = Date | number> {
     this.syncApi = api
   }
 
+  /**
+   * Add a composite index to the database
+   * @param model
+   * @param indices
+   */
+  async addModelIndex(model: StreamID, indices: Array<ModelIndex>): Promise<void> {
```

Comment:
I think the terminology throughout this PR gets really confusing regarding the difference between indexing a Model and building an index on a field or set of fields within the Model.  Can we change the names throughout to make this clearer?  For example, I think this function could be called `addFieldIndices` and the `ModelIndex` type could be called `FieldIndex`, or something like that